includes/class-shortcodes.php

```diff
@@ -492,6 +481,43 @@ private function process_category_atts( $atts, &$query_args ) {
 		);
 	}
 
+	/**
+	 * Process tag query attributes.
+	 * This checks if the `tag` or `tags` attributes are passed in the shortcode and includes them in the loop query.
+	 *
+	 * @param array $atts Shortcode attributes.
+	 * @param array $query_args The query args used to search based on attributes.
+	 *
+	 * @since x.x
+	 */
+	private function process_tag_atts( $atts, &$query_args ) {
+		if ( ! isset( $atts['tag'] ) && ! isset( $atts['tags'] ) ) {
+			return;
+		}
+
+		$requested_tags = array();
+
+		if ( isset( $atts['tag'] ) && $atts['tag'] ) {
```

Comment:
Now that you point it out... It looks like this is duplicate code. This combines tag and tags, and the other combines category and categories. Maybe we can simplify this?

Reply:
Changes made, combined the check into one reusable function

Reply:
Thanks! It looks like we could make this even less repetitive if we merge them before exploding. Like:
```
$tag = isset( $atts[ $type ] ) ? $atts[ $type ] : '';
$tags = isset( $atts[ $types ] ) ? $atts[ $types ] : '';
$sep = $tag && $tags ? ',' : '';
$tags = $tag . $sep . $tags;
```

Reply:
Thanks :) I have implemented further simplifications on the function

Reply:
Thanks. That doesn't look quite right though. The idea here is to combine 'category' and 'categories'. But the way this was changed would add a trailing or leading comma.

Reply:
Pushed an updated to remove trailing and leading commas. Run a few tests and it the results returned are correct. Not sure how to further simplify it without affecting the expected results

Reply:
This looks like a lot of extra code for this. Was my example not working?

Reply:
The example you shared would not show any listings if no category or tag is passed as a parameter. There would need to be more conditional `if` `else` statements to check if one of the parameters is set. I will make more changes to simplify it

Reply:
Maybe I'm missing something. My example sets both then combines them. If both have a value, it adds a ,

Reply:
Added a `return` if both are not set. The code works now